crates/turborepo-lib/src/config/turbo_config.rs

```diff
@@ -0,0 +1,581 @@
+    } else {
+        None
+    };
+
+    // Process preflight
+    let preflight = if let Some(preflight) = output_map.get("preflight").cloned() {
+        match preflight.as_str() {
+            "0" => Some(false),
+            "1" => Some(true),
+            _ => {
+                return Err(ConfigError::Anyhow(anyhow!(
+                    "TURBO_PREFLIGHT should be either 1 or 0."
+                )))
+            }
+        }
+    } else {
+        None
+    };
+
+    // Process enabled
+    let enabled = if let Some(enabled) = output_map.get("enabled").cloned() {
+        match enabled.as_str() {
+            "0" => Some(false),
+            "1" => Some(true),
+            _ => {
+                return Err(ConfigError::Anyhow(anyhow!(
+                    "TURBO_REMOTE_CACHE_ENABLED should be either 1 or 0."
+                )))
+            }
+        }
+    } else {
+        None
+    };
+
+    // Process timeout
+    let timeout = if let Some(timeout) = output_map.get("timeout").cloned() {
+        Some(timeout.parse::<u64>().map_err(|e| {
+            ConfigError::Anyhow(anyhow!(
+                "TURBO_REMOTE_CACHE_TIMEOUT: error parsing timeout. {}",
+                e
+            ))
+        })?)
+    } else {
+        None
+    };
+
+    let output = ConfigurationOptions {
+        api_url: output_map.get("api_url").cloned(),
+        login_url: output_map.get("login_url").cloned(),
+        team_slug: output_map.get("team_slug").cloned(),
+        team_id: output_map.get("team_id").cloned(),
+        token: output_map.get("token").cloned(),
+
+        // Processed booleans
+        signature,
+        preflight,
+        enabled,
+
+        // Processed numbers
+        timeout,
+    };
+
+    Ok(output)
+}
+
+fn get_override_env_var_config(
+    environment: &HashMap<String, String>,
+) -> Result<ConfigurationOptions, ConfigError> {
+    let mut vercel_artifacts_mapping = HashMap::new();
+    vercel_artifacts_mapping.insert(String::from("vercel_artifacts_token"), "token");
+    vercel_artifacts_mapping.insert(String::from("vercel_artifacts_owner"), "team_id");
+
+    let mut output_map = HashMap::new();
+
+    // Process the VERCEL_ARTIFACTS_* next.
+    vercel_artifacts_mapping.iter().for_each(|(k, mapped)| {
+        let k = k.to_string();
+        if let Some(value) = environment.get(&k) {
+            if !value.is_empty() {
+                output_map.insert(mapped.to_string(), value.to_string());
+            }
+        }
+    });
+
+    let output = ConfigurationOptions {
+        api_url: None,
+        login_url: None,
+        team_slug: None,
+        team_id: output_map.get("team_id").cloned(),
+        token: output_map.get("token").cloned(),
+
+        signature: None,
+        preflight: None,
+        enabled: None,
+        timeout: None,
+    };
+
+    Ok(output)
+}
+
+impl TurborepoConfigBuilder {
+    pub fn new(base: &CommandBase) -> Self {
+        Self {
+            repo_root: base.repo_root.to_owned(),
```

Comment:
tiny nit: this doesn't have to be owned, you can thread the lifetime through if you want

Reply:
It ends up used in lots of places to the point that it makes sense for the config builder to have an owned version.